bom/generate_warg_bom.py

```diff
@@ -1,49 +1,92 @@
-        if "part_num" not in line:
-            print "line doesn't have part number!" , line
-        elif line["part_num"] in found_parts:
-            #add the extra information from the component library
-            output_dict[found_parts.index(line["part_num"])].update(line)
-
-    #write the csv
-    with open(output_filename, 'wb') as output_file:
-	    dict_writer = csv.DictWriter(output_file, output_dict[0].keys())
-	    dict_writer.writeheader()
-	    dict_writer.writerows(output_dict)
-    print "wrote to {} file.".format(output_filename)
-
-if __name__ == "__main__":
-    main()
+import pip
+
+from xml.etree import ElementTree as ET
+
+import sys, os
+import csv
+
+def main():
+    if(len(sys.argv) != 3):
+        print "Make sure there are only two arguments, the netlist xml & the output name"
+    filepath, netlist_xml, output_filename = sys.argv
+
+    output_filename += ".csv"
+    
+    #found_parts is for seeing if you already added that part
+    #output_dict is the dictionary that gets written to the csv file
+    found_parts = []
+    output_dict = []
+    keylist = ['ref', 'quantity']
+
+    #Parse the netlist and store it in an lxml tree
+    tree = ET.parse(netlist_xml)
+
+    #Get a list of all the keys for the CSV at the end
+    for part_element in tree.findall(".//*[@name='part_num']...."):
+        if part_element.tag == 'comp':
+            fields = part_element.find('fields')
+            for child in fields:
+                if child.attrib['name'] not in keylist:
+                    keylist.append(child.attrib['name'])
+
+
+    #Traverse over every element with a WARG Part Number 
+    for part_element in tree.findall(".//*[@name='part_num']...."):
+        #Check if the element is under a component or a library element
+        if part_element.tag == 'comp':
+            part = dict()
+            fields = part_element.find('fields')
+            for child in fields.findall('field'):
+                part.update({child.attrib['name']:child.text})
+
+            ref = part_element.attrib['ref']
+            
+            #Find a matching part to add a quantity
+            matching_part = {}
+            for check in found_parts:
+                #Check if they both have the same keys
+                if check.keys() == part.keys():
+                    #Check if all the values are the same
+                    for key in check.keys():
+                        if check[key] != part[key]:
+                            break
+                        matching_part = check
+
+            if matching_part == {}:      
+                found_parts.append(part.copy())
+                part.update({'quantity':1, 'ref':ref})
+                output_dict.append(part)
+            else:
+                output_dict[found_parts.index(part)]['quantity'] += 1
+                output_dict[found_parts.index(part)]['ref'] += (', ' + ref)
+    reader = csv.DictReader(open('../component_library.csv', 'rb'))

```

Comment:
Still not working on my machine... maybe us the os library like I originally had.